Documentation: Updated links to landing pages for the respective notifiers in alerting

sunb3am opened the pull request:
Updated the links for notifiers supported by Grafana to their respective available landing pages.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

-->

**What this PR does / why we need it**: Redirects to landing pages of the listed supported notifiers as it should

**Which issue(s) this PR fixes**: Missing ease of access to supported notifiers

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->

Fixes # 

**Special notes for your reviewer**:

sunb3am committed:
Update _index.md

Updated the links for notifiers supported by Grafana to their respective available landing pages.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=51864) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=51864) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=51864) it.</sub>

sunb3am committed:
Documentation: Updated links to landing pages for the respective notifiers in alerting

Documentation: Updated links to landing pages for the respective notifiers in alerting, fixed linting.

sunb3am commented:
Will create new PR with fixed linting + milestones/changelog tags
